src/api/network/asio/Imp.cpp

```diff
@@ -164,30 +159,24 @@ auto Asio::Imp::data_callback(zmq::Message&& in) noexcept -> void
     if (0 == body.size()) { return; }
 
     try {
-        const auto work = [&] {
-            try {
-
-                return body.at(0).as<OTZMQWorkType>();
-            } catch (...) {
+        WorkType workType{};
+        try {
+            workType = static_cast<WorkType>(body.at(0).as<OTZMQWorkType>());
+        } catch (...) {
+            throw std::runtime_error{"Wrong size for work frame"};
```

Comment:
I know this comes from the original, but this is wrong. The function is declared noexcept, so instead of propagating the original and useful exception from .as(), it will terminate.
Change the function to noexcept(false). Also, consider allowing the useful and informative exception propagate directly from here, which would allow proper and direct initialization of 'workType'.